common-lib/src/main/scala/com/gu/mediaservice/model/ThrallMessage.scala

```diff
@@ -0,0 +1,88 @@
+package com.gu.mediaservice.model
+
+ import com.gu.mediaservice.lib.logging.{GridLogging, LogMarker}
+ import com.gu.mediaservice.model.leases.MediaLease
+ import com.gu.mediaservice.model.usage.UsageNotice
+ import org.joda.time.{DateTime, DateTimeZone}
+ import play.api.libs.json.{JodaReads, JodaWrites, Json, __}
+
+sealed trait ThrallMessage extends GridLogging with LogMarker {
+  implicit val yourJodaDateReads = JodaReads.DefaultJodaDateTimeReads.map(d => d.withZone(DateTimeZone.UTC))
+  implicit val yourJodaDateWrites = JodaWrites.JodaDateTimeWrites
+  def subject: () => String = () => this.getClass.getSimpleName
+  val id: String
+  val lastModified: DateTime
+  def additionalMarkers: () => Map[String, Any] = () => Map()
+
+  override def markerContents = {
+    val message = Json.stringify(Json.toJson(this)(ThrallMessage.writes))
+    Map (
+      "subject" -> subject,
+      "id" -> id,
+      "size" -> message.getBytes.length,
+      "length" -> message.length
+    )
+  }
+}
+object ThrallMessage{
+  implicit val yourJodaDateReads = JodaReads.DefaultJodaDateTimeReads.map(d => d.withZone(DateTimeZone.UTC))
+  implicit val yourJodaDateWrites = JodaWrites.JodaDateTimeWrites
```

Comment:
are these needed in that they're in the sealed trait, might be worth seeing if you can delete either these lines or the two lines in the trait

Reply:
it's a bit of a mess, trying to cleanly source these implicits and this wasn't needed. have added a method to the trait that json's it, which allows the convertors to live in the traits companion object rather than spamming the file